doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,805 @@
+### 3.3.1 MCLAG interface flap handling
+
+In the current implementation, in normal operation, a port isolation filter is applied on both peers to ensure that traffic received from the peer-link is never forwarded to local MCLAG member ports. This ensures loop-free and packet duplication-free operation. Then, if all ports in an MCLAG on one peer go down (MCLAG PortChannel is down), this filter is removed on the other peer, allowing traffic destined to the MCLAG to cross the peer link, maintaining connectivity to the MCLAG. Later, when at least one member port comes back up (MCLAG PortChannel is up) on the first peer, then the filter must be re-applied on the other peer. However, this can cause transient loops if the MCLAG member ports come up before the filter is re-applied on the remote peer.
+
+In this enhancement, acknowledgements are added to interface state notifications in ICCP, and these are used to ensure that the remote filter is applied before enabling traffic on the upcoming MCLAG member interface(s), thereby eliminating the possibility of loops and duplicates.
+
+### 3.3.2 MCLAG configuration change handling
+ICCP is extended to dynamically handle changes in MCLAG configurations. In the current implementation, MCLAG config attributes local ip, peer ip, peer interface and MCLAG interface configuration has to be done prior to bringing up ICCP docker. To handle configuration changes, MclagSyncd subscribes to CONFIG_DB MCLAG changes, saves configuration data in internal cache and notifies ICCPd. In this enhancement, Pre-provisioning support for MCLAG interfaces and peer interfaces is added.
+
+
+### 3.3.3 FDB handling changes
+ICCPd will be extended to support static MAC addresses, data structure changes, data type changes and MAC syncing optimizations. More details are captured in the below sections.
+
+#### 3.3.3.1 Static MAC handling
+Changes will be added to extend the current FDB TLV to notify with a static MAC flag. ICCP MAC processing will be extended for local and remote static MAC addresses handling. Local and remote static MAC move scenarios will be handled.
+
+#### 3.3.3.2 Data structure and data type modifications
+In current ICCPd implementation MAC entries are stored in linked lists which can be costly in MAC scale scenarios for lookups. The existing linked list structures will be modified to binary trees for better add, delete and lookup operations. The MAC address entry is stored as 32 byte string in the messaging and local cache, the address will be converted to 6 bytes to optimize the space messages size while syncing the MAC addresses between MCLAG peer nodes.
+
+#### 3.3.3.3 MAC sync optimizations
+When MCLAG node learns a new MAC address via ICCP from peer MCLAG node ( MAC learned on remote orphan port ), MAC address is added locally and an age notification is sent back to peer MCLAG node. These age notifications are not required to be sent, additional checks are added to stop sending such unwanted MAC updates.
```

Comment:
Why the age notification is sent back to the MCLAG node? The main MCLAG HLD doesn't mention this: https://github.com/Azure/SONiC/blob/master/doc/mclag/Sonic-mclag-hld.md#724-mac-sync-up-between-mc-lag-peers If something is not covered in the original HLD could you please add more detailed explanation here?

Reply:
@vsenchyshyn: Thanks for the comments provided. This is implementation specific detail not captured in in original HLD while the original intention of having was not clear.  The enhancements captures as the behavior is been modified and do not see the extra notification which is verified. Hope it clarifies. Thanks